### PR TITLE
#13127 fix name if not changed

### LINKED ISSUE #13127: change default connection name to hostname
When we create a new connection to the database server, the default connection name is database name
So, let's say in Oracle it is like SID it doesn't mean anything, doesn't do understanding on which server i connect. Or in postgresql i usually connect to postgres database.
Having many database server i connect on each server once, then I choose a specific database from top menu
**Describe the solution you'd like**
it would be more convenient to set default connection name to hostname of database server

### PR DESCRIPTION
**Pull request by Destrolaric**
Closes #13127

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionPageGeneral.java [hunk 1]
```diff
@@ -291,8 +290,10 @@ public void createControl(Composite parent)
             String connectionName = dataSourceDescriptor == null ? "" : dataSourceDescriptor.getName(); //$NON-NLS-1$
             connectionNameText = UIUtils.createLabelText(miscGroup, CoreMessages.dialog_connection_wizard_final_label_connection_name, CommonUtils.toString(connectionName));
             connectionNameText.addModifyListener(e -> {
-                connectionNameChanged = true;
-                ConnectionPageGeneral.this.getContainer().updateButtons();
+                if (dataSourceDescriptor == null || !connectionNameText.getText().equals(connectionName)) {
+                    connectionNameChanged = true;
+                    ConnectionPageGeneral.this.getContainer().updateButtons();
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionPageGeneral.java]:
```suggestion
                    getContainer().updateButtons();
```

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionPageGeneral.java [hunk 2]
```diff
@@ -562,8 +563,16 @@ public void saveSettings(DBPDataSourceContainer dataSource) {
...
+        String name;
+        if (dataSourceDescriptor == null) {
+            name = connectionNameChanged
+                    ? generateConnectionName(getWizard().getPageSettings(), connectionNameText.getText()) :
+                    generateConnectionName(getWizard().getPageSettings(), ModelPreferences.getPreferences()
+                            .getString(ModelPreferences.DEFAULT_CONNECTION_NAME_PATTERN));
+        } else {
+            name = connectionNameChanged ? generateConnectionName(getWizard().getPageSettings(),
+                    connectionNameText.getText()) : dataSourceDescriptor.getName();
+        }
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionPageGeneral.java]: 1. Please don't mix `if` statements with ternary operator expressions.
2. Continuation indentation must be four spaces, not 8.

**Review by ShadelessFox — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionPageGeneral.java [hunk 3]
```diff
@@ -562,8 +563,17 @@ public void saveSettings(DBPDataSourceContainer dataSource) {
...
+        String defaultValue;
+        if (dataSourceDescriptor == null) {
+            defaultValue = ModelPreferences.getPreferences()
+            .getString(ModelPreferences.DEFAULT_CONNECTION_NAME_PATTERN);
+        } else {
+            defaultValue = dataSourceDescriptor.getName();
+        }
+        String name = connectionNameChanged ? generateConnectionName(getWizard().getPageSettings(),
+            connectionNameText.getText()) :
+            generateConnectionName(getWizard().getPageSettings(), defaultValue);
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionPageGeneral.java]: 🤦🏻‍♂️

```suggestion
    final String name;

    if (connectionNameChanged) {
        name = connectionNameText.getText();
    } else if (dataSourceDescriptor != null) {
        name = dataSourceDescriptor.getName();
    } else {
        name = ModelPreferences.getPreferences().getString(ModelPreferences.DEFAULT_CONNECTION_NAME_PATTERN);
    }

    dataSource.setName(generateConnectionName(getWizard().getPageSettings(), name));
```

**Review by ShadelessFox — CHANGES_REQUESTED** [review]

**Review by ShadelessFox — APPROVED** [review]

**Comment by DariaMarkaryan** [comment]
1. project.path doesn't work
2. where does the {server} value comes from
3. we can't use {day}, {month}, {year} variables, but they should be available

**Comment by Matvey16** [comment]
Verified

**Pull request merged**